src/proto/single/ent.rs

```diff
@@ -67,13 +84,198 @@ impl JobBuilder {
     ///
     /// Sets `unique_until` on the Job's custom hash to `success`, which is Faktory's default.
     /// Another job with the same kind-args-queue combination will be accepted by the Faktory server
-    /// after the period specified in [`unique_for`](struct.JobBuilder.html#method.unique_for) has finished
+    /// after the period specified in [`unique_for`](JobBuilder::unique_for) has finished
     /// _or_ after this job has been been ***successfully*** processed.
     pub fn unique_until_success(&mut self) -> &mut Self {
         self.add_to_custom_data("unique_until", "success")
     }
 }
 
+// Ref: https://github.com/contribsys/faktory/wiki/Ent-Tracking#notes
+/// Job's state as last known by the Faktory server.
+#[derive(Debug, Clone, Deserialize)]
+#[serde(rename_all = "lowercase")]
+pub enum JobState {
+    /// The server can't tell a job's state.
+    ///
+    /// This happens when a job with the specified ID has never been enqueued, or the job has
+    /// not been marked as trackable via "track":1 in its custom hash, or the tracking info on this
+    /// job has simply expired on the server (normally, after 30 min).
+    Unknown,
+
+    /// The job has been enqueued.
+    Enqueued,
+
+    /// The job has been consumed by a worker and is now being executed.
+    Working,
+
+    /// The job has been executed with success.
+    Success,
+
+    /// The job has finished with an error.
+    Failed,
+
+    /// The jobs has been consumed but its status has never been updated.
+    Dead,
+}
+
+impl Display for JobState {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        use JobState::*;
+        let s = match self {
+            Unknown => "unknown",
+            Enqueued => "enqueued",
+            Working => "working",
+            Success => "success",
+            Failed => "failed",
+            Dead => "dead",
+        };
+        write!(f, "{}", s)
+    }
+}
+
+/// Info on job execution progress (retrieved).
+///
+/// The tracker is guaranteed to get the following details: the job's id (though they should
+/// know it beforehand in order to be ably to track the job), its last known [`state`](JobState), and
+/// the date and time the job was last updated. Additionally, arbitrary information on what's going
+/// on with the job ([`desc`](ProgressUpdate::desc)) and the job's completion percentage
+/// ([`percent`](ProgressUpdate::percent)) may be available, if the worker has provided those details.
+#[derive(Debug, Clone, Deserialize)]
+pub struct Progress {
```

Comment:
This file is getting pretty long. I wonder if we should start splitting things into submodules for individual features (like `mod progress`). Please do that in a separate PR though so we avoid holding this one up longer :+1:

Reply:
Sure! The only thing is we are done with Ent Faktory features :smile:

The rest of the enterprise features:
- [Ent Cron](https://github.com/contribsys/faktory/wiki/Ent-Cron)
- [Ent Metrics](https://github.com/contribsys/faktory/wiki/Ent-Metrics) 
- [Ent Redis Gateway](https://github.com/contribsys/faktory/wiki/Ent-Redis-Gateway)
- [Ent Throttling](https://github.com/contribsys/faktory/wiki/Ent-Throttling)
- [Ent Remote Redis](https://github.com/contribsys/faktory/wiki/Ent-Remote-Redis)

are all deployment related, with most of them fine-tuned via config files in `/etc/faktory/conf.d` on the machine.

So I suggest we split it when they introduce new features?

Reply:
... although I hope new features will be for the community edition :smile:

Reply:
I think we can probably split even now given how long the file is. But a weaker preference for sure if we're already basically done :p